server/src/test/java/org/opensearch/gateway/ReplicaShardBatchAllocatorTests.java

```diff
@@ -692,6 +745,87 @@ private RoutingAllocation onePrimaryOnNode1And1Replica(AllocationDeciders decide
         );
     }
 
+    private RoutingAllocation routingAllocationWithMultipleShards(AllocationDeciders deciders, int primaries, int replicas)
+        throws InterruptedException {
+        return routingAllocationWithMultipleShards(deciders, Settings.EMPTY, primaries, replicas, UnassignedInfo.Reason.CLUSTER_RECOVERED);
+    }
+
+    private RoutingAllocation routingAllocationWithMultipleShards(
+        AllocationDeciders deciders,
+        Settings settings,
+        int primaries,
+        int replicas,
+        UnassignedInfo.Reason reason
+    ) throws InterruptedException {
+        Map<ShardId, ShardRouting> shardIdShardRoutingMap = new HashMap<>();
+        Index index = null;
+        for (ShardId shardIdFromBatch : shardsInBatch) {
+            shardIdShardRoutingMap.put(
+                shardIdFromBatch,
+                TestShardRouting.newShardRouting(shardIdFromBatch, node1.getId(), true, ShardRoutingState.STARTED)
+            );
+            if (index == null) {
+                index = shardIdFromBatch.getIndex();
+            }
```

Comment:
Isn't this always for index `test` ?

Reply:
Yes, currently index can only be```test``` but avoiding the hard-coding helps in keeping the method more generic.